cmd/state/main.go

```diff
@@ -260,15 +260,18 @@ func run(args []string, isInteractive bool, cfg *config.Instance, out output.Out
 	return err
 }
 
-func argsHaveVerbose(args []string) bool {
-	for _, arg := range args {
+// Returns copy of args with any "-v" or "--verbose" flag removed, and true if such a flag was removed.
+func removeVerboseArg(args []string) ([]string, bool) {
+	isRunOrExec := len(args) > 1 && (args[1] == "run" || args[1] == "exec")
```

Comment:
This will detect the run or exec commands only when they are the first argument. If another flag is set as the first argument, then the command will not be detected. This increases my thought that this issue should be resolved by altering the flag handling rather than post-processing the args.